src/bindgen/bindings.rs

```diff
@@ -57,6 +57,27 @@ impl Bindings {
         any
     }
 
+    pub fn struct_field_names(&self, path: &BindgenPath) -> Option<Vec<String>> {
+        let item_container = self.struct_map.get_items(path)?;
+        let mut fields = Vec::<String>::new();
+        for container in item_container {
+            let mut pos: usize = 0;
+            if let ItemContainer::Struct(ref st) = container {
+                for field in &st.fields {
+                    if let Some(found_pos) = fields.iter().position(|v| *v == field.0) {
```

Comment:
This seems it could get pretty slow... I'll try to do some performance measurement...

Off hand, if the only use-case for this so far is bitflags, I'd prefer to just special-case the code generation for single-field structs, which would be much simpler of a patch.

Reply:
Alternatively maybe this should be memoized / cached somehow. This does a bunch of work for each constant of a given type, specially if there are a lot of fields.

Reply:
I found this bug during use bitflags. But, already there are other test cases. And, the actual reason is MSVC doesn't support C11 and designated initializer is not a part of C++11 ([It will be added to C++2a](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2017/p0329r4.pdf)). IMO, Using list-initializer for every C++ case can resolve this error clearly.

As your suggestion, I'll add some memoizing codes.